### PR TITLE
Convert Ursula PR tests to use a Heat provisioner scheme

### PR DESCRIPTION
**Pull request by yangzhang-ibm**
### Problem resolved:
Currently Ursula PR tests are setup using the Ansible os_* module to provision nodes and other resources.  This approach to provisioning and deployment orchestration for test environments is too cumbersome and not a good fit for creating multiple test environments.

### Solution of this commit:
This commit is to add the Heat provisioning capability to the Ursula PR tests and move what we currently have off of Ansible os_* modules onto Heat. Heat is decent at resource provisioning and deployment orchestration and defining environments is much cleaner with the use of templates.

### Some explanations for my code changes:
* the tmp/ folder is required by ursula-cli to store the generatd ssh_config and ssh_key file, without creating that folder in advance, ursula-cli will raise an error;
* I removed the "--sudo" arg from ursula commands in test/setup and test/run scripts, that is to avoid duplication since ursula-cli will always add that when heat provisioner is being used.
* I have put the fixed content of envs/example/ci-full/group_vars/all.yml into that file directly, and let test/setup script only write the dynamic content during runtime with an appending mode instead of previous overwriting mode.

### CODE REVIEW

File: playbooks/ci-full/templates/heat_stack.yml [hunk 1]
```diff
@@ -0,0 +1,230 @@
+heat_template_version: 2013-05-23
```
Comment (j2sol) [inline playbooks/ci-full/templates/heat_stack.yml]: Where is this date coming from?

**Comment by j2sol** [comment]
ok to test

**Comment by j2sol** [comment]
I think this is okay

**Comment by j2sol** [comment]
retest

File: playbooks/ci-full/templates/heat_stack.yml (hunk 1 above)
Comment (yangzhang-ibm) [inline playbooks/ci-full/templates/heat_stack.yml]: This version is from the template in sitecontroller code. And that is a valid version number mentioned in heat HOT specification: http://docs.openstack.org/developer/heat/template_guide/hot_spec.html#id1.
I think that version can fulfill our requirement, if we encounter an issue with the version in the future, we can consider change to a later one then. It works well for now.
fyr, HOT versions have these options: 2013-05-23, 2014-10-16, 2015-04-30, etc

**Comment by yangzhang-ibm** [comment]
@j2sol Thanks for your review and comments, fyr, the PR test for this will only succeed after my another PR for jjb gets merged. https://github.blueboxgrid.com/blue-box-cloud/jenkins-job-builder/pull/59

**Comment by j2sol** [comment]
retest

**Comment by timrchavez** [comment]
I submitted PR https://github.com/blueboxgroup/ursula-cli/pull/48

**Comment by timrchavez** [comment]
FWIW we set OS_PROJECT_ID in our stackrc on the Jenkins slave.  We don't sent OS_PROJECT_NAME or OS_TENANT_NAME.

**Comment by timrchavez** [comment]
We put support for OS_PROJECT_NAME into our stackrc's and I've updated the jenkins slaves, so all that remains is the change to ursula-cli to look for it to merge.

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
Yay!

**Pull request merged**